mongo/datadog_checks/mongo/common.py

```diff
@@ -36,27 +36,43 @@ def get_state_name(state):
         return 'UNKNOWN'
 
 
+def get_long_state_name(state):
+    """Maps a mongod node state id to a human readable string."""
+    if state in REPLSET_MEMBER_STATES:
+        return REPLSET_MEMBER_STATES[state][1]
+    else:
+        return 'Replset state %d is unknown to the Datadog agent' % state
+
+
 class Deployment(object):
-    def get_available_metrics(self):
-        # TODO: Use this method to know what metrics to collect based on the deployment type.
+    def is_principal(self):
+        """Whether or not this node can be used to collect data from collections. Should only
+        be True for one node in the cluster."""
         raise NotImplementedError
 
 
 class MongosDeployment(Deployment):
-    def get_available_metrics(self):
-        return None
+    def is_principal(self):
+        # A mongos has full visibility on the data, Datadog agents should only communicate
+        # with one mongos.
+        return True
 
 
 class ReplicaSetDeployment(Deployment):
-    def __init__(self, replset_name, replset_state):
+    def __init__(self, replset_name, replset_state, in_shard=False):
         self.replset_name = replset_name
         self.replset_state = replset_state
         self.replset_state_name = get_state_name(replset_state).lower()
+        self.in_shard = in_shard
+        self.is_primary = replset_state == 1
 
-    def get_available_metrics(self):
-        return None
+    def is_principal(self):
+        # There is only ever one primary node in a replica set.
+        # In case sharding is disabled, the primary can be considered the master.
+        return not self.in_shard and self.replset_state == 1
```

Comment:
Did you mean to use the `.is_primary` attribute added just above?

```suggestion
        return not self.in_shard and self.is_primary
```